pkgs/build-support/php/hooks/composer-install-hook.sh

```diff
@@ -0,0 +1,75 @@
+declare -a composerVendorCache
+declare -a installComposerDevDependencies
+declare -a installComposerScriptsDependencies
+declare -a installComposerPluginsDependencies
+declare -a dontComposerInstallConfigureHook
+declare -a dontComposerInstallBuildHook
+declare -a dontComposerInstallInstallHook
+
+if [[ -z "${dontComposerInstallConfigureHook-}" ]]; then
+  preConfigureHooks+=(composerInstallConfigureHook)
+fi
+
+if [[ -z "${dontComposerInstallBuildHook-}" ]]; then
+  preBuildHooks+=(composerInstallBuildHook)
+fi
+
+if [[ -z "${dontComposerInstallInstallHook-}" ]]; then
+  preInstallHooks+=(composerInstallInstallHook)
+fi
+
+composerInstallConfigureHook() {
+    echo "Executing composerInstallConfigureHook"
+
+    if [[ ! -e "${composerVendorCache}" ]]; then
+        echo "No composer cache found."
+        exit 1
+    fi
+
+    cp -fr ${composerVendorCache}/. .
+    chmod +w .composer -R
+
+    echo "Finished composerInstallConfigureHook"
+}
+
+composerInstallBuildHook() {
+    echo "Executing composerInstallBuildHook"
+
+    argstr=("--no-interaction" "--optimize-autoloader")
+
+    if [[ ! -n ${installComposerDevDependencies-} ]]; then
+        argstr+=("--no-dev")
+    fi
+
+    if [[ ! -n ${installComposerScriptsDependencies-} ]]; then
+        argstr+=("--no-scripts")
+    fi
+
+    if [[ ! -n ${installComposerPluginsDependencies-} ]]; then
+        argstr+=("--no-plugins")
+    fi
+
+    argstr+=("--apcu-autoloader-prefix=nixPredictableApcuAutoloaderPrefix")
+
+    COMPOSER_DISABLE_NETWORK=1 \
+    COMPOSER_CACHE_DIR=.composer \
+    composer install ${argstr[@]}
```

Comment:
In case we add an argument containing spaces:

```suggestion
    composer install "${argstr[@]}"
```

Reply:
fixed